src/Gremlin.Net.CosmosDb/ISchemaBoundTraversal.Schema.Extensions.cs

```diff
@@ -432,6 +432,13 @@ private static string GetPropertyName(Type elementType, LambdaExpression lambda)
             if (elementType != propInfo.ReflectedType && !elementType.IsSubclassOf(propInfo.ReflectedType))
                 throw new ArgumentException($"Expression '{lambda}' refers to a property that is not from type {elementType}.");
 
+            var jsonProperty = propInfo.GetCustomAttributes(typeof(Newtonsoft.Json.JsonPropertyAttribute), false).OfType<Newtonsoft.Json.JsonPropertyAttribute>().FirstOrDefault();
+            if (jsonProperty != null)
+            {
+              if (jsonProperty.PropertyName != null && jsonProperty.PropertyName != "")
```

Comment:
Minor suggestion to simplify this:

```suggestion
              if (!String.IsNullOrWhiteSpace(jsonProperty.PropertyName))
```